src/sagemaker/model.py

```diff
@@ -1114,7 +1114,7 @@ def prepare_container_def(self, instance_type=None, accelerator_type=None):
     def _upload_code(self, key_prefix, repack=False):
         """Placeholder Docstring"""
         local_code = utils.get_config_value("local.local_code", self.sagemaker_session.config)
-        if self.sagemaker_session.local_mode and local_code:
+        if (self.sagemaker_session.local_mode and local_code) or self.entry_point is None:
```

Comment:
What's this code doing? If a customer uses a training script whole training, would this entry_point be None when we do estimator.deploy